### ISSUE TITLE
Error: TimeoutError: Pa11y timed out

### ISSUE THREAD

**Issue by renedet** [pa11y/pa11y#401]
## Expected behaviour
**A simple pa11y test does not work these sites:**
http://www.tjpe.jus.br
https://www.trt8.jus.br

## Actual behaviour
**When I execute the commands:**
pa11y -s WCAG2A http://www.tjpe.jus.br
pa11y -s WCAG2A https://www.trt8.jus.br

**Show this error:**

```
Welcome to Pa11y

 > Running Pa11y on URL https://www.trt8.jus.br

Error: TimeoutError: Pa11y timed out (30000ms)
    at Timeout.setTimeout [as _onTimeout] (/home/renedet/.nvm/versions/node/v9.8.0/lib/node_modules/pa11y/node_modules/p-timeout/index.js:27:54)
    at ontimeout (timers.js:466:11)
    at tryOnTimeout (timers.js:304:5)
    at Timer.listOnTimeout (timers.js:267:5)
```

**I had downloaded the HTML from the sites, to try local with files, because I thought it was a problem with the connection or routing, but it did not work.
Another sites work fine.**

**I had change time out option "-t" to value 120000, but not work again.**

I had try another option to check accessibility [IDI Web Accessibility Checker](https://achecker.ca/checker/index.php) and work well with http://www.tjpe.jus.br and https://www.trt8.jus.br

Websites http://www.tjpe.jus.br and https://www.trt8.jus.br are online and appear work fine in browser.

## Steps to reproduce
**Just run this command:**
pa11y -s WCAG2A http://www.tjpe.jus.br
pa11y -s WCAG2A https://www.trt8.jus.br

## Environment:
Pa11y:      5.0.3
Node.js:    9.8.0
npm:        6.0.0
OS:         4.4.0-121-generic (linux)

**Comment by rowanmanning** [comment]
Hi,

Just looking into this, and I think I've nearly found a cause. Pa11y uses Puppeteer under the hood, and [the `networkidle2` option here](https://github.com/GoogleChrome/puppeteer/blob/v1.3.0/docs/api.md#pagegotourl-options). This means that the Pa11y tests don't run until:
> there are no more than 2 network connections for at least 500 ms
I'm testing to see whether there are any long-running network connections on those pages. Do you know if an analytics script or similar might be making constant requests?

**Comment by rowanmanning** [comment]
OK this is actually to do with some SSL issues on those pages. I'm trying to find out how we can make Pa11y itself handle these, but in the meantime there are 3 URLs on the `http://www.tjpe.jus.br` page which never load correctly and so the page just hangs:

```
https://www.tjpe.jus.br/documents/10180/1251212/BANNER+BUSCA+ATIVA.png/fce03a06-75fa-4e7a-88ef-5a2404a7eb6a?t=1464370632926
https://www.tjpe.jus.br/documents/10180/1861114/banner+custas+e+emolumentos/196381d1-17ba-39ac-a138-e17e29173de7?t=1518172627484
https://www.tjpe.jus.br/documents/10180/1856579/Banner_home_coleta_seletiva+%281%29.png/f1dac4a7-91a7-0976-1317-7e7428b3fe43?t=1517586299510
```

There are two possible workarounds for you:

1. Make those resources use `http` instead of `https` as the certificate is invalid
2. Add a `pa11y.json` file to your working directory which has the following (I'm not actually certain why this fixes it yet):
    ```json
    { "chromeLaunchConfig": { "ignoreHTTPSErrors": false } }
    ```

**Comment by renedet** [comment]
Thanks Rowan!

Nice! Your config `pa11y.json` works to me.

**Comment by asaranprasad** [comment]
I am facing TimeoutError while using the tool for the first time, even when used over a static html file. 
Is there any first-time configuration that I miss?

Thanks for your help!

![image](https://user-images.githubusercontent.com/14305132/41936578-8e352ee0-795b-11e8-8eca-f62b81e9985e.png)

**Comment by asaranprasad** [comment]
Forget it. My bad. :D Path problem

**Comment by naveen7036** [comment]
> OK this is actually to do with some SSL issues on those pages. I'm trying to find out how we can make Pa11y itself handle these, but in the meantime there are 3 URLs on the `http://www.tjpe.jus.br` page which never load correctly and so the page just hangs:
> 
> ```
> https://www.tjpe.jus.br/documents/10180/1251212/BANNER+BUSCA+ATIVA.png/fce03a06-75fa-4e7a-88ef-5a2404a7eb6a?t=1464370632926
> https://www.tjpe.jus.br/documents/10180/1861114/banner+custas+e+emolumentos/196381d1-17ba-39ac-a138-e17e29173de7?t=1518172627484
> https://www.tjpe.jus.br/documents/10180/1856579/Banner_home_coleta_seletiva+%281%29.png/f1dac4a7-91a7-0976-1317-7e7428b3fe43?t=1517586299510
> ```
> There are two possible workarounds for you:
> 
> 1. Make those resources use `http` instead of `https` as the certificate is invalid
> 2. Add a `pa11y.json` file to your working directory which has the following (I'm not actually certain why this fixes it yet):
>    ```json
>    { "chromeLaunchConfig": { "ignoreHTTPSErrors": false } }
>    ```

Where should we give this pa11y.json, do we need to create it and give, I tried so, but no help of it.
I am getting the error: SSL_Cert_Authority inspite of export and import of cert files to my java folder.
And manually able to load the pages after the export, import of cert files. But getting the above error through pa11y. Any help is much appreciated.

**Comment by sowmyasridharamurthy** [comment]
> Thanks Rowan!
> 
> Nice! Your config `pa11y.json` works to me.

Hi Rowan, 

pa11y.json did not work for me, 
I am using the same URL's mentioned above by renedet. 

I created a new project folder and simply followed the steps mentioed in pa11y installation. 

is there more configurations that needs to be done?

**Comment by josebolos** [comment]
Hi @sowmyasridharamurthy 

This issue is closed, would you mind creating a new Issue with all the necessary details?

Thanks!
